packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -1005,11 +1024,12 @@ export class IndexedDbPersistence implements Persistence {
 
   private detachWindowUnloadHook(): void {
     if (this.windowUnloadHandler) {
+      assert(this.window != null, "'window' should be defined");
```

Comment:
We only use `debugAssert` and `hardAssert` and not the asserts from Node. This should be a `debugAssert` since this would be an SDK bug (versus an external bug from a hardAssert). I do think you can combine it with the assert below though:

```
debugAssert(typeof this.window?.removeEventListener === 'function',	    ...   );
```